app/models/card.rb

```diff
@@ -16,16 +16,44 @@ class Card < ActiveRecord::Base
 
   def check_translation(user_translation)
     if full_downcase(translated_text) == full_downcase(user_translation)
-      update_attributes(review_date: Time.now + 3.days)
+      set_review_date_for_step
     else
       false
     end
   end
 
+  def reset_review_step
+    if review_attempt < 3
+      update_attributes(review_attempt: [review_attempt + 1, 3].min)
+    else
+      update_attributes(review_step: 1, review_attempt: 1)
+    end
+  end
+
   protected
 
-  def set_review_date
-    self.review_date = Time.now + 3.days
+  def set_review_date_as_now
+    self.review_date = Time.now
+  end
+
+  def set_review_date_for_step
```

Comment:
Assignment Branch Condition size for set_review_date_for_step is too high. [21.59/15]